pyccel/ast/core.py

```diff
@@ -197,6 +197,18 @@ def _eval_subs(self, old, new):
     def _eval_evalf(self,prec):
         return sp_Pow(self.base,self.exp).evalf(prec)
 
+    @property
+    def is_real(self):
+        return self._args[0].is_real and self._args[1].is_real
+
+    @property
+    def is_integer(self):
+        return self._args[0].is_integer and self._args[1].is_integer
+
+    @property
+    def is_complex(self):
+        return self._args[0].is_complex and self._args[1].is_complex
```

Comment:
See above comment

Reply:
the same thing if it's an integer or real => it's a complex